amf-webapi/shared/src/main/scala/amf/plugins/document/webapi/parser/spec/jsonschema/ReferenceResolver.scala

```diff
@@ -0,0 +1,45 @@
+package amf.plugins.document.webapi.parser.spec.jsonschema
+
+import amf.plugins.document.webapi.parser.spec.common.YMapEntryLike
+import org.yaml.model.YMap
+import amf.core.parser.YMapOps
+import scala.util.Try
+import java.net.URI
+import scala.annotation.tailrec
+
+trait ReferenceResolver {
+  def resolve(reference: String, index: Map[String, YMapEntryLike]): Option[YMapEntryLike]
+}
+
+object FragmentTraversingResolver extends ReferenceResolver {
+
+  override def resolve(reference: String, index: Map[String, YMapEntryLike]): Option[YMapEntryLike] =
+    Try(new URI(reference)).toOption.flatMap { uri =>
```

Comment:
I remember I tech-talk with Nacho Manzano where he talked about some performance caveats with scala `Try` instead of try catch. Don't remember concretely what the problem was though.